Remove flaky E2E test from current set

<!--------- For bugs and general issues --------->
**Setup**
- Rancher version: 2.6.5
- Browser type & version: CI

**Describe the bug**
We currently have several blocking errors generated by the E2E, preventing to merge or slowing it down, as the test needs to be restarted.

**To Reproduce**
Create PR to master.

**Result**
https://github.com/rancher/dashboard/actions/workflows/e2e.yaml?query=is%3Afailure

**Expected Result**
No errors or test successful. 

**Screenshots**
-

**Additional context**
-

We have 3 different cases in 2 tests.

#### 1.
"Home Page > Displays changelog section" 
tests/pages/home.spec.ts
```
CypressError: Timed out retrying after 4050ms: 'cy.click()' failed because this element is detached from the DOM.
'<li data-v-3fcb6f81="" class="user-menu-item">...</li>'
``` 
([e.g.](https://github.com/rancher/dashboard/runs/6870791206?check_suite_focus=true#step:6:114))

```
AssertionError: Timed out retrying after 4000ms: Expected to find element: `.user-menu-item`, but never found it. Queried from element: <div#page-actions.actions>
```
([e.g.](https://github.com/rancher/dashboard/runs/6877432194?check_suite_focus=true#step:6:130))

#### 2.
"Side navigation: Cluster > Can close first menu groups on click" 
tests/navigation/side-nav/product-side-nav.spec.ts
```
AssertionError: Timed out retrying after 4000ms: Too many elements found. Found '1', expected '0'.
```
 ([e.g.](https://github.com/rancher/dashboard/runs/6896172277?check_suite_focus=true#step:6:251))

--- a/cypress/integration/tests/pages/home.spec.ts
+++ b/cypress/integration/tests/pages/home.spec.ts
@@ -17,7 +17,7 @@ describe('Home Page', () => {
     burgerMenuPo.checkVisible();
   });
 
-  it('Displays changelog section', () => {
+  it.skip('Displays changelog section', () => {
     const homePage = new HomePagePo();
     const pageActionsPo = new PageActions();
 

